packages/curve-ui-kit/src/hooks/useSortFromQueryString.ts

```diff
@@ -0,0 +1,29 @@
+import { useLocation, useNavigate } from 'react-router'
+import { useCallback, useMemo } from 'react'
+import { SortingState, OnChangeFn } from '@tanstack/react-table'
+
+export function useSortFromQueryString(defaultSort: SortingState) {
+  const { search } = useLocation()
+  const navigate = useNavigate()
+  const sort = useMemo(() => parseSort(search, defaultSort), [defaultSort, search])
+  const onChange: OnChangeFn<SortingState> = useCallback(
+    (newSort) => navigate({ search: updateSort(search, typeof newSort == 'function' ? newSort(sort) : newSort) }),
+    [navigate, search, sort],
+  )
+  return [sort, onChange] as const
+}
+
+function parseSort(search: string, defaultSort: SortingState) {
+  const sort = new URLSearchParams(search)
+    .getAll('sort')
+    .map((id) => (id.startsWith('-') ? { id: id.slice(1), desc: true } : { id, desc: false }))
+  return sort.length ? sort : defaultSort.map(({ id, desc }) => ({ id: id.replace(/\./g, '_'), desc }))
+}
+
+export function updateSort(search: string, state: SortingState): string {
+  const params = new URLSearchParams(search)
+  params.delete('sort')
+  state.forEach(({ id, desc }) => params.append('sort', `${desc ? '-' : ''}${id}`))
```

Comment:
This works for one column, but for now multi-column sorting will not work.

Reply:
Multi-column sorting works fine, you need to hold the shift when clicking on the 2nd column though (default tanstack behavior). For example I get `/#/ethereum/beta-markets?sort=-rates_borrowApyPcent&sort=rates_lendApyPcent`

Reply:
Huh, I wonder who else knows about this...

Reply:
yeah it's not obvious, we'd need to provide better user support. But I also dont like the idea to enable multisort by default